beRajeevKumar/Frontend_Mentor
# Contact form solution

# Pull Request: Enhanced Contact Form Implementation
### Live demo solution [Netlify solution](https://contact-form-valid.netlify.app/)
## What does this PR do?
This pull request introduces an enhanced version of the contact form, focusing on improving accessibility, responsiveness, and user experience. The solution aligns closely with the provided design specifications and incorporates several new features aimed at making the form more user-friendly.

Fixes #147 

### Motivation and Context
The goal of this implementation is to create a contact form that not only looks good but also provides a seamless experience for all users, including those with disabilities. By improving accessibility and user experience, we aim to ensure that everyone can interact with the form without difficulty.

### Dependencies
No additional dependencies are required for this change.

### Screenshots
![desktop-error](https://github.com/user-attachments/assets/583bde36-b2d3-433e-82f4-29562b5eece9)
![mobile-solution](https://github.com/user-attachments/assets/6516b880-f79e-4226-9a12-f0a77af32d88)


## Type of change

## How should this be tested?
To verify the changes, please follow these steps:
1. Check the accessibility features using a screen reader.
2. Fill out the form to test real-time validation for all input fields.
3. Submit the form and ensure the success message appears and disappears after 5 seconds.
4. Refresh the page to check if the saved draft is pre-filled in the form.

### Test Cases
- [ ] Test the accessibility of the form with various screen readers.
- [ ] Validate required fields and ensure dynamic error messages display correctly.
- [ ] Check the responsiveness of the form on different screen sizes.


## Request for Merge
@beRajeevKumar  I kindly request a review of this pull request and look forward to your feedback. If everything looks good, I would appreciate merging it into the main branch. Thank you for your time and support!
